core-java-modules/core-java-collections-maps-8/src/test/java/com/baeldung/map/castingmaptoobject/CastingMapToObjectUnitTest.java

```diff
@@ -0,0 +1,68 @@
+package com.baeldung.map.castingmaptoobject;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import org.apache.commons.beanutils.BeanUtils;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CastingMapToObjectUnitTest {
+
+    private static final Map<String, Object> map = Map.of(
+        "id", 1L,
+        "name", "Baeldung",
+        "addresses", List.of(
+                new Address("La Havana", new Country("Cuba")),
+                new Address("Paris", new Country("France"))
+        )
+    );
+
+    @Test
+    void givenMap_whenUsingBeanUtils_thenConvertToObject() throws InvocationTargetException, IllegalAccessException {
+        User user = new User();
+        BeanUtils.populate(user, map);
+
+        assertEquals(user.getId(), map.get("id"));
+        assertEquals(user.getName(), map.get("name"));
+        assertEquals(user.getAddresses(), map.get("addresses"));
```

Comment:
The first argument of assertEquals should be the expected value. Same in other tests.

Plus, since the assertion is exactly the same in all tests, how about extracting a helper method to make it clear that each test is proving the same thing?